README_DLC.md

```diff
@@ -0,0 +1,189 @@
+
+```bash
+sudo apt update -y 
+sudo apt install software-properties-common -y  
+sudo add-apt-repository ppa:deadsnakes/ppa -y
+sudo apt-get install python3.10-tk -y  
+sudo apt install python3.10 python3.10-dev -y 
+sudo apt install python3.10-distutils -y
+```
+
+Create an environment called e.g. dlcenv
+
+```bash
+mkdir -p ~/Documents/PYTHON/envs
+cd ~/Documents/PYTHON/envs
+python3.10 -m venv dlcenv
+```
+
+Activate the environment and install packages
+
+```bash
+source ~/Documents/PYTHON/envs/dlcenv/bin/activate
+pip install setuptools==65
+pip install ibllib
+pip install torch==1.12
+pip install deeplabcut[tf]
+```
+
+### Test if tensorflow and deeplabcut installation was successful
+
+Export environment variables for testing
+```bash
+CUDA_VERSION=11.8
+export PATH=/usr/local/cuda-$CUDA_VERSION/bin:$PATH
+export TF_FORCE_GPU_ALLOW_GROWTH='true'
+export LD_LIBRARY_PATH=/usr/local/cuda-$CUDA_VERSION/lib64:/usr/local/cuda-$CUDA_VERSION/extras/CUPTI/lib64:$LD_LIBRARY_PATH  
+```
+
+Try to import deeplabcut and tensorflow (don't forget that dlcenv has to be active)
+```
+python -c 'import deeplabcut, tensorflow'
+```
+
+Once the import goes through without errors (it is ok to get the warning that you cannot use the GUI), you can set up an alias in your .bashrc file to easily enter the dlcenv environment:
+```bash
+nano ~/.bashrc
+```
+Enter this line under the other aliases:
+```bash
+alias dlcenv="CUDA_VERSION=11.8; export PATH=/usr/local/cuda-%CUDA_VERSION/bin:$PATH; export TF_FORCE_GPU_ALLOW_GROWTH='true'; export LD_LIBRARY_PATH=/usr/local/cuda-$CUDA_VERSION/lib64:/usr/local/cuda-$CUDA_VERSION/extras/CUPTI/lib64:$LD_LIBRARY_PATH; source ~/Documents/PYTHON/envs/dlcenv/bin/activate"
+```
+After opening a new terminal you should be able to type `dlcenv` and end up in an environment in which you can import tensorflow and deeplabcut like above.
+
+### Clone and install iblvideo
+
+Make sure to be in the Documents/PYTHON folder and that the dlcenv environment is activated
+```bash
+cd ~/Documents/PYTHON
+dlcenv
+```
+Then clone and install iblvideo
+```
+git clone https://github.com/int-brain-lab/iblvideo.git
+cd iblvideo
+pip install -e .
+```
+
+Test if you install was successful
+```
+python -c 'import iblvideo'
+```
+
+Eventually run the tests:
+```shell
+pytest ./iblvideo/tests/test_pose_dlc.py
+pytest ./iblvideo/tests/test_motion_energy.py
+```
+
+Note that some variables are routinely computed from DLC output in the IBL ephys pipeline, such as pupil diameter and lick times. See ibllib/pipes/ephys_preprocessing.py EphysPostDLC and brainbox.behavior.dlc for more details.
+
+## Releasing a new version (for devs)
+
+We use semantic versioning, with a prefix: `vMAJOR.MINOR.PATCH`. If you update the version, see below for what to adapt.
+
+### Any version update
+Update the version in
+```
+iblvideo/iblvideo/__init__.py
+```
+Afterwards, tag the new version on Github.
+
+
+### Update MINOR or MAJOR
```

Comment:
Do you think we want to decouple the DLC weights from the iblvideo version number also? Looking at the code it seems you have already done this, i.e using the __dlc__version=v2.2. Can this be updated in the read me also